unit/src/DependencyResolverTest.C

```diff
@@ -144,3 +144,21 @@ TEST_F(DependencyResolverTest, dependsOnTest)
   EXPECT_FALSE(_tree.dependsOn("m1", "something_else"));
   EXPECT_FALSE(_tree.dependsOn("something_else", "k0"));
 }
+
+TEST_F(DependencyResolverTest, cyclicTest)
+{
+  try
+  {
+    // Attempt to insert a dependency that results in cyclicity
+    _resolver.insertDependency("a", "b");
+    FAIL() << "missing expected exception";
+  }
+  catch (const std::exception & e)
```

Comment:
No, the only change needed is 
`catch (const CyclicDependencyException<T> & e)`, where `T` is the type of your resolver. (`std::string`?) - That'll only mean you recompile this unit test.